ion/tokenizer.go

```diff
@@ -1369,6 +1369,77 @@ func (t *tokenizer) unread(c int) {
 	t.buffer = append(t.buffer, c)
 }
 
+// CheckNullAnnotation reads elements in the stream and checks if it is a null type annotation (e.g., 'null.string::').
+// Returns an error if the read was unsuccessful or it was a null type annotation. Returns nil otherwise.
+func (t *tokenizer) checkNullAnnotation() error {
+	skipped, err := t.SkipDot()
+	if err != nil {
+		return err
+	}
+
+	if skipped {
+		var ret []int
+
+		hasElementInStream := true
+		for hasElementInStream {
+			var c int
+			c, err = t.read()
+			if err != nil {
+				hasElementInStream = false
+				break
+			}
+			if !isIdentifierPart(c) {
+				hasElementInStream = false
+				t.unread(c)
+				break
+			}
+			if c == -1 {
+				err = io.EOF
+				hasElementInStream = false
+				break
+			}
+			ret = append(ret, c)
+		}
+
+		// Skips potential whitespace in annotation. e.g., 'null.string  ::'
+		hasWhiteSpaceInStream := true
+		for hasWhiteSpaceInStream {
+			var c int
+			c, err = t.read()
+			if err != nil {
+				hasWhiteSpaceInStream = false
+				break
+			}
+			if c != ' ' {
```

Comment:
This is going to get more complex here. space is not the only whitespace that can go between the symbol and the double colon right? I have have tabs for example, and I think new-lines etc. 

I think we discussed this offline and though a nice feature if this is getting more and more complex I would drop it for now and make an issue for it to come back to it later.

Reply:
Removed, and created an [issue](https://github.com/amzn/ion-go/issues/162) for this